File: backend/app/routes/copilot.py
from fastapi import APIRouter, Depends

from app.schemas.copilot import (
    SuggestReplyRequest,
    SuggestReplyResponse,
    SummarizeRequest,
    SummarizeResponse,
    KBRetrieveRequest,
    KBRetrieveResponse,
)
from app.core.security import get_current_user
from app.services import copilot_service

router = APIRouter()


@router.post("/suggest-reply", response_model=SuggestReplyResponse)
async def suggest_reply(
    body: SuggestReplyRequest,
    current_user: dict = Depends(get_current_user),
):
    result = await copilot_service.suggest_reply(
        tenant_id=current_user["tenant_id"],
        ticket_id=body.ticket_id,
        conversation_id=body.conversation_id,
        context=body.context,
    )
    return SuggestReplyResponse(**result)


@router.post("/summarize", response_model=SummarizeResponse)
async def summarize(
    body: SummarizeRequest,
    current_user: dict = Depends(get_current_user),
):
    result = await copilot_service.summarize(
        tenant_id=current_user["tenant_id"],
        ticket_id=body.ticket_id,
        conversation_id=body.conversation_id,
    )
    return SummarizeResponse(**result)


@router.post("/retrieve-kb", response_model=KBRetrieveResponse)
async def retrieve_kb(
    body: KBRetrieveRequest,
    current_user: dict = Depends(get_current_user),
):
    snippets = await copilot_service.retrieve_kb_snippets(
        tenant_id=current_user["tenant_id"],
        query=body.query,
        top_k=body.top_k,
    )
    return KBRetrieveResponse(snippets=snippets)

File: frontend/services/copilot.ts
import { api } from "@/lib/api";
import type { ApiResponse, SuggestedReply, TicketSummary, KBSnippet } from "@/types";

export const copilotService = {
  async suggestReply(
    ticketId: string,
    conversationId: string
  ): Promise<ApiResponse<SuggestedReply>> {
    return api.post("/copilot/suggest-reply", {
      ticket_id: ticketId,
      conversation_id: conversationId,
    });
  },

  async summarize(
    ticketId: string,
    conversationId: string
  ): Promise<ApiResponse<TicketSummary>> {
    return api.post("/copilot/summarize", {
      ticket_id: ticketId,
      conversation_id: conversationId,
    });
  },

  async retrieveKB(query: string): Promise<ApiResponse<KBSnippet[]>> {
    return api.post("/copilot/retrieve-kb", { query });
  },
};
